rust/lance-arrow/src/lib.rs

```diff
@@ -412,6 +417,12 @@ impl RecordBatchExt for RecordBatch {
         let struct_array: StructArray = self.clone().into();
         self.try_new_from_struct_array(project(&struct_array, schema.fields())?)
     }
+
+    fn take(&self, indices: &UInt32Array) -> Result<RecordBatch> {
```

Comment:
Does this preserve metadata? Might want to use `try_new_from_struct_array`

Reply:
Fixed